nix-web: 0.1.0 -> 0.2.0

liclac opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

https://codeberg.org/gorgon/gorgon/releases/tag/nix-web-v0.2.0

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

liclac committed:
nix-web: 0.1.0 -> 0.2.0

liclac commented:
This also pins the package to use Nix 2.17 until it can be made 2.18 compatible. Applying the backport label as Nix 2.18 became the default in 23.11.

RaitoBezarius commented on pkgs/by-name/ni/nix-web/package.nix:
```diff
@@ -3,24 +3,26 @@
 , openssl
-, nix
+, nixVersions
+, nixVersion ? "nix_2_17"
```
This virtually prevent from passing any arbitrary Nix version for test, I find it a bit sad.
Could we have something like `{ nixVersions, nix ? nixVersions.nix_2_17 }` rather ? More than often, a user will be able to pass the full package directly.

liclac replied:
I actually tried doing this at first, but an input called `nix` gets passed the default nix version, so I started doing this... but actually, I can just call it `nixPackage ? nixVersions.nix_2_17`. How's that?

Mindavi replied:
Seems fine to me for now.

RaitoBezarius replied:
Yeah, let's revisit this later!

liclac replied:
Whoops, I forgot to actually make this change

RaitoBezarius replied:
Feel free to send a follow up PR, I didn't want you to be blocked on those considerations :)

kloenk committed:
nix-web: Build on platforms.unix, depend on OpenSSL if not on Darwin

This matches the `platform` setting for nix itself, as much as I don't
have a good way to test this on anything but Linux right now.

Co-Authored-By: Finn Behrens <me@kloenk.de>

Mindavi reviewed (changes requested):
Build is broken on darwin.

liclac commented:
@Mindavi @Kloenk fixed macOS ^^

kloenk reviewed (approved):
LGTM

build on `aarch64-darwin` and `aarch64-linux`

RaitoBezarius commented:
This has sitted for too long, apologies!
